### PR TITLE
gnu-complexity 1.13

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gnu-complexity.rb [hunk 1]
```diff
@@ -21,12 +21,25 @@ class GnuComplexity < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "6bc40505bf964f2ac7ef30d2f65c8180832e709c49cf6872b8651caf6a84b1a1"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
   depends_on "autogen"
 
+  # Fix build problem in doc. Borrowed from Debian.
```
Comment (SMillerDev) [inline Formula/gnu-complexity.rb:28]: Has this been upstreamed?

Comment (ZhongRuoyu) [reply Formula/gnu-complexity.rb:28]: No. `master` and `v1.13` are at the same commit: https://git.savannah.nongnu.org/cgit/complexity.git

Comment (SMillerDev) [reply Formula/gnu-complexity.rb:28]: Could you submit the fix upstream then?

Comment (ZhongRuoyu) [reply Formula/gnu-complexity.rb:29]: Done. Reported both issues to project maintainer @brkorb via email, as I couldn't find another way.

Comment (brkorb) [reply Formula/gnu-complexity.rb:29]: The author/maintainer has no idea how to incorporate this into "upstream".
gnu-complexity.rb is not part of the sources. :)

**Review by brkorb — COMMENTED** [review]
BTW: bruce.korb@gmail.com or bkorb@gnu.org

**Comment by ZhongRuoyu** [comment]
> The author/maintainer has no idea how to incorporate this into "upstream".
> gnu-complexity.rb is not part of the sources. :)

`gnu-complexity.rb` is a formula in Homebrew that defines how the package should be built. Essentially, there are two issues that need to be addressed. Now that @brkorb is here, let me just post my email here:

<details>

1. When configured with the `--disable-dependency-tracking` option, the following error was emitted (output truncated):

```plaintext
Making all in doc
make[2]: Entering directory '/home/ruoyu/complexity-1.13/doc'
make[2]: *** No rule to make target '.deps', needed by 'example.texi'.  Stop.
make[2]: Leaving directory '/home/ruoyu/complexity-1.13/doc'
make[1]: *** [Makefile:1016: all-recursive] Error 1
make[1]: Leaving directory '/home/ruoyu/complexity-1.13'
make: *** [Makefile:957: all] Error 2
```

If I understand this correctly, this is because `DEPDIR` has already been predefined, so the redefinition at [`doc/Makefile.am:45`](https://git.savannah.gnu.org/cgit/complexity.git/tree/doc/Makefile.am#n45) did not take effect. Notice the following error while running `autoreconf -i`:

```plaintext
doc/Makefile.am:45: warning: DEPDIR was already defined in condition TRUE, which includes condition !AMDEP ...
configure.ac:19: ... 'DEPDIR' previously defined here
```

Debian maintainer @bbonev has come up with a fix [here](https://salsa.debian.org/debian/complexity/-/raw/69a7b9d27eb5c2ba8aa43966518971df74d55657/debian/patches/01_fix_autobuild.patch). You may want to have a look at it.

2. Also, the files `src/opts.{c,h}` seem to be generated by a version of AutoGen that's newer than the latest stable release. These files are "Generated from AutoOpts 42:2:17 templates", while those generated by AutoGen's latest release 5.18.16, are from 42:1:17. As a result, the following error, among many others, was emitted during the build:

```plaintext
opts.h:62:3: error: #error option template version mismatches autoopts/options.h header
  62 | # error option template version mismatches autoopts/options.h header
     |   ^~~~~
```

These two files are generated from `src/opts.def`, and can only be found in the tarball. Here are the first few lines of `src/opts.h`:

```plaintext
/*   -*- buffer-read-only: t -*- vi: set ro:
 *
 *  DO NOT EDIT THIS FILE   (opts.h)
 *
 *  It has been AutoGen-ed
 *  From the definitions    opts.def
 *  and the template file   options
 *
 * Generated from AutoOpts 42:2:17 templates.
 *
 *  AutoOpts is a copyrighted work.  This header file is not encumbered
 *  by AutoOpts licensing, but is provided under the licensing terms chosen
```

Were these files generated by an unstable version of AutoGen? Running `cd src && autogen opts.def` with AutoGen 5.18.16 fixes the issue, and that is already being done on at least [Debian](https://salsa.debian.org/debian/complexity/-/blob/master/debian/rules) and [Arch (AUR)](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=complexity).

</details>

**Comment by brkorb** [comment]
Probably the right choice now is to disable the dependencies on derived (generated) sources.
Over the years, the method of developing "configure" scripts kept changing and autogen
has a gazillion derived files. So, when I was actively maintaining it, I just adapted to changes.
It got revamped and all the configury stuff needs a full rethink and rewrite. Until that happens,
autogen will likely not be released again. For the record, 90-95% of all my effort went into
configure scripts and I learned to hate and despise them, so I confess to being less than eager.
I'm also retired and prefer photography to this. That doesn't help, either.

BTW, removing dependencies is probably not a good idea anyway. Rebuilding the docs requires
running the "complexity" program itself and feeding the help text back into the document.

**Comment by ZhongRuoyu** [comment]
Thanks for your efforts, @brkorb. I could tell that it probably wasn't a pleasant experience dealing with the `configure` scripts; I spent quite some time on the source repo yesterday, but I still failed to generate all the stuffs in the tarball, and ended up working directly on the tarball instead.

As per @brkorb's feedback, I think we can stick to the current changes, then. We patch a minor issue in `doc/Makefile.am`, and regenerate `src/opts.{c,h}` from `src/opts.def`.

File: Formula/gnu-complexity.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/gnu-complexity.rb:29]: It's about this patch though: https://salsa.debian.org/debian/complexity/-/raw/69a7b9d27eb5c2ba8aa43966518971df74d55657/debian/patches/01_fix_autobuild.patch

Comment (bbonev) [reply Formula/gnu-complexity.rb:29]: If you're improving the patches in some way, please keep me in the loop

**Review by p-linnane — APPROVED** [review]

**Pull request closed without merging**